File: backend/emm-sale/src/main/java/com/emmsale/event/domain/EventStatus.java
package com.emmsale.event.domain;

import static com.emmsale.event.exception.EventExceptionType.INVALID_STATUS;

import com.emmsale.event.exception.EventException;
import java.util.Arrays;
import lombok.Getter;

@Getter
public enum EventStatus {

  IN_PROGRESS("진행 중"),
  UPCOMING("진행 예정"),
  ENDED("종료된 행사");

  private final String value;

  EventStatus(final String value) {
    this.value = value;
  }

  public static EventStatus from(final String value) {
    return Arrays.stream(values())
        .filter(status -> status.isSameValue(value))
        .findFirst()
        .orElseThrow(() -> new EventException(INVALID_STATUS));
  }

  public boolean isSameValue(String value) {
    return this.value.equals(value);
  }
}

File: android/2023-emmsale/app/src/main/java/com/emmsale/data/event/EventStatus.kt
package com.emmsale.data.event

enum class EventStatus(val text: String) {
    IN_PROGRESS("진행 중"),
    SCHEDULED("진행 예정"),
    ENDED("종료된 행사");

    companion object {
        fun from(status: String): EventStatus = values().find { it.text == status }
            ?: throw IllegalArgumentException("${status}는 올바르지 않은 행사 상태입니다.")
    }
}
